### PR TITLE
docs: Rearrange all the things

### PR DESCRIPTION
**Pull request by joshka**
This commit shuffles around the world and adds a bunch of missing
documentation:

- Adds mdbook-linkcheck
- Ensures that links are valid (and fixes the ones that aren't)
- Flips the Home and Introduction pages so that the first thing someone
  sees is the introduction rather than the about this book info
- Moves how to and FAQ section before concepts
- Groups the How To and Concepts sections by topic
- Moves Features to the references section
- Adds a new Backends section to concepts
- Adds a few new header pages to make it easy to expand the sections
- Simplified a bunch of page names

Before / After:
image image

### CODE REVIEW

File: src/how-to/develop-applications/tracing.md
Comment (kdheepak) [inline src/how-to/develop-applications/tracing.md:1]: Delete this `develop-applications` folder.

File: src/tutorial/json-editor/closing_thoughts.md [hunk 1]
```diff
@@ -3,8 +3,8 @@
 This tutorial should get you started with a basic understanding of the flow of a `ratatui` program.
 However, this is only _one_ way to create a `ratatui` application. Because `ratatui` is relatively
 low level compared to other UI frameworks, almost any application model can be implemented. You can
-explore more of these in [Concepts: Application Patterns](./../../concepts/application-patterns.md)
-and get some inspiration for what model will work best for your application.
+explore more of these in [Concepts: Application Patterns](../../concepts/application-patterns) and
```
Comment (kdheepak) [inline src/tutorial/json-editor/closing_thoughts.md]: Can we point it to the readme instead?

```suggestion
explore more of these in [Concepts: Application Patterns](../../concepts/application-patterns/README.md) and
```
```

**Comment by joshka** [comment]
https://ratatui-org.github.io/ratatui-book/pr-preview/pr-61/html/ is the link to the depoloyed PR view (need to fix this up as it changes when we add linkcheck from the root)

**Comment by kdheepak** [comment]
I love the changes!

File: src/tutorial/json-editor/closing_thoughts.md (hunk 1 above)
Comment (joshka) [reply src/tutorial/json-editor/closing_thoughts.md]: yep, this does work as a link btw in case that's not obvious.

File: src/SUMMARY.md [hunk 2]
```diff
@@ -1,12 +1,13 @@
 # Summary
 
-[Home](./README.md)
+[Introduction to Ratatui](./README.md)
 
----
+# User Guide
 
-- [Introduction](./introduction.md)
+- [About this book](./introduction.md)
```
Comment (kdheepak) [inline src/SUMMARY.md]: Maybe just get rid of this page? I don't really see the purpose. It "feels off" when you compare it to the previous and next page.

Comment (kdheepak) [reply src/SUMMARY.md]: Or move this content to the first `README.md` link.

File: src/tutorial/json-editor/closing_thoughts.md (hunk 1 above)
Comment (kdheepak) [reply src/tutorial/json-editor/closing_thoughts.md]: I didn't know that that worked but I also like the local files linked exactly so I can hit a keyboard shortcut and open the exact file :)

File: src/SUMMARY.md (hunk 2 above)
Comment (joshka) [reply src/SUMMARY.md]: Agreed - I thought about doing this, but went conservative there and left it, but I agree totally with removing it.

**Comment by kdheepak** [comment]
I do think there's diminishing returns going to 3 levels in the TOC.

The official rust book never goes to 3 levels

image

**Comment by joshka** [comment]
> I do think there's diminishing returns going to 3 levels in the TOC.

We go to 4 in some of the tutorials. :D
The more important thing to notice about the rust book is that no section has more than 7 elements. which makes it easy to scan.

File: src/tutorial/json-editor/closing_thoughts.md (hunk 1 above)
Comment (joshka) [reply src/tutorial/json-editor/closing_thoughts.md]: makes sense - this is fixed

**Pull request merged**